### ISSUE TITLE
Support Linode

### ISSUE THREAD

**Issue by hbjydev** [opencost/opencost#2128]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to use OpenCost, but I use Linode, and you don't have hooks into their API as of yet.

**Describe the solution you'd like**
I'd like Linode to be supported in the same way Scaleway are, preferrably.

**Describe alternatives you've considered**
I considered creating a custom config for it, but the pricing is too granular in comparison for me to be able to figure out.

**Additional context**
https://www.linode.com/docs/api/linode-types/#types-list

**Comment by mattray** [comment]
I'm not familiar with Linode's Kubernetes offering but we'd love to support it.

**Comment by rchekhina** [comment]
HI,

I need it too and for linode plan for unmanaged kubernetes.

Best regards.

**Comment by ar4mirez** [comment]
@mattray I'll love to help on this one, if none is taking it.

**Comment by mattray** [comment]
@ar4mirez we'd definitely appreciate Linode support, feel free to open a PR, ask questions and reach out.

**Comment by ar4mirez** [comment]
> @ar4mirez we'd definitely appreciate Linode support, feel free to open a PR, ask questions and reach out.

Sounds good just wanted to make sure to no duplicate work. I'm going to allocate time to work on this over the weekend.

**Comment by 0xGermain** [comment]
@ar4mirez any work done on this ? I would love to have a Linode implementation

**Comment by ar4mirez** [comment]
@0xGermain Yes, I got a good amount of work done, but between holidays and work it's been on hold for a bit. This coming weekend I set aside time to wrap it up.

**Comment by dwbrown2** [comment]
@ar4mirez we're just wrapping up a release and should have review cycles soon. Just let us know how we can help.

**Comment by ar4mirez** [comment]
@dwbrown2 I'll be pushing my code this weekend, sorry for my delayed responses I've been swamped with work but I'm using weekend hours to wrap this up.

**Comment by mattray** [comment]
@ar4mirez just checking in, do you need any assistance getting this in?

**Comment by rchekhina** [comment]
HI,

Do you have some new ?

Best regards.

**Comment by ar4mirez** [comment]
@mattray @rchekhina I do apologize for the delay. I had some personal things to take care of, but I'm back in business. I was getting close to getting this done before I had to unplug. I'll get back to this over the weekend and see if I can wrap it up.

**Comment by mattray** [comment]
@ar4mirez Sounds great, feel free to ping me directly via email or on Slack if you want to coordinate getting this into the next release.

**Comment by ar4mirez** [comment]
@mattray, I got my hands back on this issue, but I need a little more time as I am only working on it on weekends. If I encounter any problems this weekend, I'll ping you on Slack. When is the next release planned? Do I need to speed this up?

**Comment by mattray** [comment]
The next release is planned for the last week of March most likely.

**Comment by tbaka-dev** [comment]
Howdy folks + @ar4mirez 
I'm popping in from the Solutions Engineering team at Akamai Connected Cloud (previously Linode). We heard you were working on this, and I need to get familiar with the repo here, but we're happy to help how we can.

**Comment by ar4mirez** [comment]
@tbaka, thanks for popping in! :D I started working on this feature and almost reached the finish line; lately, I've been at work-related events. I'll make time to get this to the finish line. If I can't get it this weekend, I'll seek help.

**Comment by AjayTripathy** [comment]
Please let me know how I can help! Eager to take a PR here :)

**Comment by mattray** [comment]
@tbaka I've got an email thread with Kevin Murray if you want to move this there. I can work with you to get this fast-tracked.

**Comment by rchekhina** [comment]
HI,

Do you have some news about the linode support please ?

Best regards.

**Comment by AjayTripathy** [comment]
@ar4mirez anything to share here? How can I help?

**Comment by ar4mirez** [comment]
I apologize for the delay. I was trying to get it done. It's been crazy for me at work. If anyone would like to take over, feel free. I've been in firefighter mode. I'm so sorry; I wanted to finish it, but it's impossible now.

**Comment by mhmxs** [comment]
> I apologize for the delay. I was trying to get it done. It's been crazy for me at work. If anyone would like to take over, feel free. I've been in firefighter mode. I'm so sorry; I wanted to finish it, but it's impossible now.

@ar4mirez i would pick this up. Could you please give me some context, what you did, what is missing, where to find the branch, etc. :pray:
